src/throttle/amoc_throttle_controller.erl

```diff
@@ -153,28 +149,40 @@ handle_call({resume, Name}, _From, State) ->
         Error -> {reply, Error, State}
     end;
 handle_call({change_rate, Name, Rate, Interval}, _From, State) ->
-    Info = maps:get(Name, State),
-    case maybe_change_rate(Name, Rate, Interval, Info) of
-        {ok, Rate} -> UpdatedInfo = Info#throttle_info{rate = Rate, interval = Interval},
-            {reply, ok, State#{Name => UpdatedInfo}};
-        Error -> {reply, Error, State}
+    case State of
+        #{Name := Info} ->
+            case maybe_change_rate(Name, Rate, Interval, Info) of
+                {ok, Rate} ->
```

Comment:
Right, this line is matching on it giving the provided value, if there's any diff this one doesn't match and the call will return that other rate that had been calculated.

Reply:
damn, so `{ok, _}` is patternpamtched as Error clause...
this line should be changed to:
`{ok, NewRate} -> UpdatedInfo = Info#throttle_info{rate = NewRate, interval = Interval},`

Reply:
I don't think we need to store here the rate as returned by `maybe_change_rate`, that one is the rate given to each worker, which is `Rate div NoOfProcesses`. As it is currently, `throttle_info` will contain the newly provided rate, I think current code is already correct, have a look at the latest commit :)